analytics/client.py

```diff
@@ -216,13 +219,17 @@ def _enqueue(self, msg):
         if not self.send:
             return True, msg
 
-        try:
-            self.queue.put(msg, block=False)
-            self.log.debug('enqueued %s.', msg['type'])
+        if self.max_queue_size == 0: # If we're not using a queue, send immediately (synchronously)
```

Comment:
Instead of this, could we use a queue with a size of 1, and block when we add items to it?

```
self.queue = queue.Queue(1)
...
self.queue.put(msg, block=true)
```

Reply:
(The question wasn't addressed to me, but I don't think the original author will have time for it. Is it possible to move forward anyway?)

It seems to me that you have a consumer thread in mind (otherwise blocking is forever and pointless), while the purpose of these changes (#101) was to make the library work where using threading isn't a good option.